tools/check_for_upgrade.sh

```diff
@@ -29,9 +29,9 @@ fi
 # Cancel upgrade if git is unavailable on the system
 whence git >/dev/null || return 0
 
-if [ -f ~/.zsh-update ]
+if [ -f ${ZSH_CACHE_DIR}.zsh-update ]
```

Comment:
Should probably have a `/` here, like `${ZSH_CACHE_DIR}/.zsh-update`. We probably shouldn't assume `$ZSH_CACHE_DIR` to have a trailing slash, since it may be set by the user.